pyproject.toml

```diff
@@ -12,42 +12,68 @@ authors = [
 requires-python = "==3.12.*"
 license = { text = "BSD-3-Clause" }
 dependencies = [
-  "appdirs>=1.4",
-  "azure-core>=1.26",
-  "azure-identity>=1.16.1",
-  "azure-keyvault-certificates>=4.6",
-  "azure-keyvault-keys>=4.6",
-  "azure-keyvault-secrets>=4.6",
-  "azure-mgmt-automation>=1.0",
-  "azure-mgmt-compute>=30.3",
-  "azure-mgmt-containerinstance>=10.1",
-  "azure-mgmt-dns>=8.0",
-  "azure-mgmt-keyvault>=10.3",
-  "azure-mgmt-msi>=7.0",
-  "azure-mgmt-network>=25.0",
-  "azure-mgmt-rdbms>=10.1",
-  "azure-mgmt-resource>=23.0",
-  "azure-mgmt-storage>=21.1",
-  "azure-storage-blob>=12.15",
-  "azure-storage-file-datalake>=12.10",
-  "azure-storage-file-share>=12.10",
-  "chevron>=0.14",
-  "cryptography>=42.0",
-  "fqdn>=1.5",
-  "msal>=1.21",
-  "psycopg>=3.1",
-  "pulumi>=3.80",
-  "pulumi-azure-native>=2.49.1",
-  "pulumi-random>=4.14",
-  "pulumi-tls>=5.0.3",
-  "pydantic>=2.4",
-  "pyjwt>=2.8",
-  "pytz>=2023.3",
-  "PyYAML>=6.0",
-  "rich>=13.4",
-  "simple-acme-dns>=3.0",
-  "typer>=0.9",
-  "websocket-client>=1.5",
+  "appdirs==1.4.4",
+  "azure-core==1.30.2",
+  "azure-identity==1.17.1",
+  "azure-keyvault-certificates==4.8.0",
+  "azure-keyvault-keys==4.9.0",
+  "azure-keyvault-secrets==4.8.0",
+  "azure-mgmt-compute==32.0.0",
+  "azure-mgmt-containerinstance==10.1.0",
+  "azure-mgmt-dns==8.1.0",
+  "azure-mgmt-keyvault==10.3.1",
+  "azure-mgmt-msi==7.0.0",
+  "azure-mgmt-rdbms==10.1.0",
+  "azure-mgmt-resource==23.1.1",
+  "azure-mgmt-storage==21.2.1",
+  "azure-storage-blob==12.22.0",
+  "azure-storage-file-datalake==12.16.0",
+  "azure-storage-file-share==12.17.0",
+  "chevron==0.14.0",
+  "cryptography==43.0.0",
+  "fqdn==1.5.1",
+  "psycopg==3.2.1",
+  "pulumi-azure-native==2.56.0",
+  "pulumi-random==4.16.3",
+  "pulumi==3.129.0",
+  "pydantic==2.8.2",
+  "pyjwt[crypto]==2.9.0",
+  "pytz==2024.1",
+  "pyyaml==6.0.2",
+  "rich==13.7.1",
+  "simple-acme-dns==3.1.0",
+  "typer==0.12.4",
+  "websocket-client==1.8.0",
+]
+
+[project.optional-dependencies]
+docs = [
+  "emoji==2.12.1",
+  "myst-parser==4.0.0",
+  "pydata-sphinx-theme==0.15.4",
+  "sphinx-togglebutton==0.3.2",
+  "sphinx==8.0.2",
+]
+lint = [
+  "ansible-dev-tools==24.7.2",
+  "ansible==10.2.0",
+  "black==24.4.2",
+  "mypy==1.10.1",
+  "pandas-stubs==2.2.2.240807",
+  "pydantic==2.8.2",
+  "ruff==0.5.0",
+  "types-appdirs==1.4.3.5",
+  "types-chevron==0.14.2.20240310",
+  "types-pytz==2024.1.0.20240417",
+  "types-pyyaml==6.0.12.20240311",
+  "types-requests==2.32.0.20240622",
+]
```

Comment:
The same here, I don't _think_ the core dependencies are needed for the linting scripts.

Reply:
As above - this won't install the core dependencies.